.gitignore

```diff
@@ -12,3 +12,42 @@
 
 # Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736
 .glide/
+
+
+# Created by https://www.gitignore.io/api/go,git,vim
+
+### Git ###
+*.orig
+
+### Go ###
+# Binaries for programs and plugins
+*.exe
+*.dll
+*.so
+*.dylib
+
+# Test binary, build with `go test -c`
+*.test
+
+# Output of the go coverage tool, specifically when used with LiteIDE
+*.out
+
+# Project-local glide cache, RE: https://github.com/Masterminds/glide/issues/736
+.glide/
+
+### Vim ###
+# swap
+[._]*.s[a-v][a-z]
+[._]*.sw[a-p]
+[._]s[a-v][a-z]
+[._]sw[a-p]
+# session
+Session.vim
+# temporary
+.netrwhist
+*~
+# auto-generated tag files
+tags
+
+# End of https://www.gitignore.io/api/go,git,vim
+
```

Comment:
might not be bad to add `vendor` here so any internal vendor directories don't get added

Reply:
so my understanding of vendoring is that we'd want to check in all the vendored code into our repository if we decide to vendor?  Is the thinking here that since we're not vendoring we shouldn't allow checking in of vendored code?

Reply:
It is a fair point that we shouldn't decide on this until we adopt a vendoring strategy, you are right - so let's leave this as it is.

> so my understanding of vendoring is that we'd want to check in all the vendored code into our repository

Traditionally both models are commonly used - checked-in vendor directory, and not-checked-in vendor directory - it mostly depends on the strategy of the vendor tool being used. For example, sometimes only a manifest is checked in, and then some build tool checks out dependencies based on commit SHAs in external repos - other strategies include copying all source in-tree, so you don't need to rely on those external repos keeping those commits in them.